microsoft/terminal
# ci: update to Spell check to 0.0.17a

## Summary of the Pull Request
Upgrade the spell checking action

 
## References
https://github.com/microsoft/terminal/pull/9004#issuecomment-771825083


## PR Checklist


## Detailed Description of the Pull Request / Additional comments

### Plurals and paste tenses
In the past, plurals `foo`+`s` and past tenses `foo`+`ed` were automatically tolerated. This turned out to be a bad design choice on my part.

The basic example is that `potatos` would sometimes be treated as a mistake and sometimes not (depending on the presence of `potato`).

You can see in this PR, that this logic resulted in `Applys` being accepted as a word along with `AppContainered` -- there's nothing intrinsically wrong w/ the latter, but unfortunately in order to screen out the former, my shortcut just couldn't stick around. This means that the `dictionary`/`expect` files will grow perhaps by a tiny bit, but as you can see, not really by much.

This is also why `thereses` (a user) was accepted as a word in the past ([therese](https://en.wikipedia.org/wiki/Th%C3%A9r%C3%A8se_of_Lisieux) is in the base dictionary, so `therese` + `s` was acceptable).

### Pull requests

When GitHub initially introduced GitHub Actions, the event for `pull_request` was created without enough permission for a tool like this to work properly. I worked around that by using the `schedule` event. In 2020, they introduced a replacement event `pull_request_target` which has enough permission. This means that I can stop relying on the `schedule` event.

### Miscellaneous

* I've folded together some `expect/` files since now is as good a time as any.
* I've included a hint about `excludes.txt` (I added a similar one for our primary repo recently, and it came up this week in `microsoft/terminal` -- @zadjii-msft)
* I've standardized on a default of `.github/actions/spelling` to make the out of the box experience easier for new adopters, so I'm applying that change here -- if you're attached to the old directory name, specifying it is still supported. -- note the directory rename may cause a merge conflict for people with open PRs and changes to the contents, this shouldn't be a big problem.


## Validation Steps Performed
I've created a temporary repo which is using this as its default branch.

There's a sample PR here:
https://github.com/check-spelling/terminal/pull/1

If people want, they're welcome to create PRs into that repo to test (you can also test in your own forks -- just be sure that you enable or re-enable the workflow -- GitHub has disabled workflows with `schedule` in inactive repositories to save CPU cycles).
